src/native/s3_client.c

```diff
@@ -340,14 +340,25 @@ JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_s3_S3Client_s3ClientMake
     jclass jni_class,
     jlong jni_s3_client,
     jobject java_s3_meta_request_jobject,
+    jbyteArray jni_region,
     jint meta_request_type,
     jbyteArray jni_marshalled_message_data,
     jobject jni_http_request_body_stream,
+    jlong jni_credentials_provider,
     jobject java_response_handler_jobject) {
     (void)jni_class;
 
     struct aws_allocator *allocator = aws_jni_get_allocator();
     struct aws_s3_client *client = (struct aws_s3_client *)jni_s3_client;
+    struct aws_credentials_provider *credentials_provider = (struct aws_credentials_provider *)jni_credentials_provider;
+    struct aws_signing_config_aws *signing_config = NULL;
+    bool success = false;
+    if (credentials_provider) {
+        struct aws_byte_cursor region = aws_jni_byte_cursor_from_jbyteArray_acquire(env, jni_region);
+        signing_config = aws_mem_calloc(allocator, 1, sizeof(struct aws_signing_config_aws));
+        aws_s3_init_default_signing_config(signing_config, region, credentials_provider);
+        aws_jni_byte_cursor_from_jbyteArray_release(env, jni_region, region);
```

Comment:
Do we need to copy the `region` before releasing?

Reply:
good catch!